crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -87,6 +87,16 @@ fn inheritance_cycle_initial<'db>(_db: &'db dyn Db, _self: Class<'db>) -> Option
     None
 }
 
+/// When looking up an attribute on a class, we sometimes need to avoid
+/// looking up attributes defined on the `object` class.
+#[derive(Clone, Debug, Copy, PartialEq, Eq)]
+pub(crate) enum MroAttributeLookupPolicy {
+    /// Exclude attributes defined on `object`
+    NoObjectFallback,
+    /// Look up all classes in MRO, including `object`
+    WithObjectFallback,
```

Comment:
nit: `object` isn't really a "fallback", it's just part of the MRO. So I would probably name these `ExcludeObject` and `Default`.